File: wagerproof-ios-native/Wagerproof/Features/Props/Components/PropPlayerCard.swift
import SwiftUI
import WagerproofModels
import WagerproofDesign

/// Player-specific prop card, styled to match the MLB game card
/// (`GameRowCard`) — same rounded surface, avatar-with-team-color-glow, and
/// right-edge trend sparkline — but on a slightly lifted dark card surface.
///
/// Layout:
///   - Main row: player headshot · player name · centered Over/Under pills ·
///     labeled trend chart.
///   - Bottom info row: BEST (the player's best market) · WHAT'S NEXT (the
///     opponent) · a Liquid Glass game-time pill in the bottom-right.
/// Tapping pushes `PlayerPropDetailView`.
struct PropPlayerCard: View {
    let item: PlayerPropFeedItem
    let namespace: Namespace.ID
    let onSelect: (PlayerPropSelection) -> Void

    @Environment(\.colorScheme) private var colorScheme

    private var sel: PlayerPropSelection { item.selection }
    private var computed: MLBPropComputedAtLine { item.headline.computed }
    private var row: MLBPlayerPropRow { item.headline.row }

    private var primary: Color { Color(hex: Int(item.teamPrimaryHex)).teamVisible(in: colorScheme) }
    private var secondary: Color { Color(hex: Int(item.teamSecondaryHex)).teamVisible(in: colorScheme) }

    /// Lifted card surface — matches the MLB dark cards but a touch lighter so
    /// the cards read clearly above the near-black feed background.
    private var cardFill: Color { Color(light: 0xFFFFFF, dark: 0x202024) }

    var body: some View {
        Button {
            onSelect(sel)
        } label: {
            content
        }
        .buttonStyle(.plain)
        .matchedTransitionSource(id: sel.transitionID, in: namespace)
        .sensoryFeedback(.impact(weight: .light), trigger: sel.id)
    }

    private var content: some View {
        let shape = RoundedRectangle(cornerRadius: 26, style: .continuous)
        return VStack(alignment: .leading, spacing: 8) {
            mainRow
            Divider().background(Color.appBorder.opacity(0.5))
            bottomInfoRow
        }
        .padding(.leading, 12)
        // Extra right padding so the trend chart sits in from the edge.
        .padding(.trailing, 14)
        .padding(.vertical, 9)
        .background {
            ZStack {
                // Matches AgentRowCard's glass treatment: ultraThinMaterial
                // thinned in dark mode so more of the page shows through.
                shape.fill(.ultraThinMaterial)
                    .opacity(colorScheme == .dark ? 0.55 : 1)
                shape.strokeBorder(Color.appBorder.opacity(0.5), lineWidth: 0.5)
            }
        }
        .clipShape(shape)
        .shadow(color: .black.opacity(0.06), radius: 4, x: 0, y: 2)
    }

    // MARK: - Main row

    private var mainRow: some View {
        HStack(alignment: .center, spacing: 10) {
            avatar
            identity
            Spacer(minLength: 8)
            overUnderBlock
            Spacer(minLength: 8)
            trendBlock
        }
    }

    // MARK: - Identity (name + vs opponent underneath)

    private var identity: some View {
        // Player name (no leading lineup number — that's a web artifact) with
        // the opponent matchup directly beneath it.
        VStack(alignment: .leading, spacing: 2) {
            Text(sel.playerName)
                .font(.system(size: 15, weight: .bold))
                .foregroundStyle(Color.appTextPrimary)
                .lineLimit(1)
                .minimumScaleFactor(0.75)
            if !sel.opponentAbbr.isEmpty {
                Text("vs \(sel.opponentAbbr)")
                    .font(.system(size: 11, weight: .medium))
                    .foregroundStyle(Color.appTextSecondary)
                    .lineLimit(1)
            }
        }
    }

    // MARK: - Avatar (player headshot + team logo badge + glow)

    private var avatar: some View {
        // The headshot disc and the team-logo badge are two close glass discs;
        // the container liquid-merges them where they overlap (iOS 26). The
        // disc carries the team-color glass tint, the badge a clear glass chip
        // so the team logo on top still reads. Pre-26 both fall back to the
        // gradient/solid discs they had before.
        LiquidGlassMergeContainer(spacing: 14) {
            ZStack(alignment: .bottomTrailing) {
                PlayerHeadshot(playerId: sel.playerId, size: 40)
                    .frame(width: 44, height: 44)
                    .teamGlassDisc(primary: primary, secondary: secondary, fallbackStroke: cardFill)
                    // One soft team halo for depth; light so it doesn't muddy
                    // the glass or the merge seam.
                    .shadow(color: primary.opacity(0.22), radius: 5, x: 0, y: 1)

                if let logo = sel.teamLogoUrl, let url = URL(string: logo) {
                    AsyncImage(url: url) { phase in
                        if case .success(let img) = phase {
                            img.resizable().scaledToFit()
                        } else {
                            Color.clear
                        }
                    }
                    .frame(width: 16, height: 16)
                    .padding(2)
                    .liquidGlassBackground(in: Circle())
                    .offset(x: 2, y: 2)
                }
            }
            .frame(width: 44, height: 44)
        }
    }

    // MARK: - Over / Under pills (centered in the empty space)

    private var overUnderBlock: some View {
        VStack(spacing: 4) {
            ouPill(prefix: "O", value: MLBPlayerProps.formatOdds(computed.overOdds), tint: Color.appPrimary)
            ouPill(prefix: "U", value: MLBPlayerProps.formatOdds(computed.underOdds), tint: Color.appTextSecondary)
        }
        .fixedSize()
    }

    private func ouPill(prefix: String, value: String, tint: Color) -> some View {
        HStack(spacing: 4) {
            Text("\(prefix) \(MLBPlayerProps.formatLine(computed.line))")
                .font(.system(size: 9, weight: .bold))
                .foregroundStyle(Color.appTextMuted)
            Text(value)
                .font(.system(size: 11, weight: .bold, design: .monospaced))
                .foregroundStyle(tint)
        }
        .lineLimit(1)
        .padding(.horizontal, 8)
        .padding(.vertical, 4)
        .background(Color.appSurfaceMuted.opacity(0.55), in: Capsule())
        .overlay(Capsule().stroke(Color.appBorder.opacity(0.6), lineWidth: 0.5))
    }

    // MARK: - Trend chart (labeled, slightly larger)

    private var trendBlock: some View {
        VStack(alignment: .trailing, spacing: 3) {
            Text("L10 TREND")
                .font(.system(size: 8, weight: .bold))
                .tracking(0.5)
                .foregroundStyle(Color.appTextMuted)
            RecentFormStrip(strip: computed.miniStrip, line: computed.line)
                .frame(width: 74, height: 46)
        }
    }

    // MARK: - Bottom info row (Best · What's next · time)

    private var bottomInfoRow: some View {
        HStack(alignment: .center, spacing: 16) {
            infoItem(label: item.metricLabel, value: MLBPlayerProps.marketLabel(row.market), valueColor: Color.appPrimary)
            // Summary stats for that market: last-10 over rate + hit rate.
            infoItem(label: "L10", value: "\(computed.l10.fractionLabel) Over", valueColor: Color.appTextPrimary)
            infoItem(label: "HIT", value: computed.l10.pctLabel, valueColor: hitColor)
            Spacer(minLength: 0)
            timePill
        }
    }

    private var hitColor: Color {
        guard let pct = computed.l10.pct else { return Color.appTextMuted }
        if pct >= 70 { return Color.appPrimary }
        if pct >= 55 { return Color(hex: 0xEAB308) }
        return Color.appTextSecondary
    }

    private func infoItem(label: String, value: String, valueColor: Color) -> some View {
        VStack(alignment: .leading, spacing: 1) {
            Text(label)
                .font(.system(size: 8, weight: .bold))
                .tracking(0.5)
                .foregroundStyle(Color.appTextMuted)
            Text(value)
                .font(.system(size: 12, weight: .semibold))
                .foregroundStyle(valueColor)
                .lineLimit(1)
        }
    }

    // MARK: - Time pill (bottom-right)

    private var timePill: some View {
        Text(MLBFormatting.gameTime(sel.gameTimeEt))
            .font(.system(size: 9, weight: .bold, design: .monospaced))
            .foregroundStyle(Color.appTextSecondary)
            .padding(.horizontal, 7)
            .padding(.vertical, 3)
            .liquidGlassBackground(in: Capsule())
            .overlay(Capsule().stroke(Color.appBorder.opacity(0.6), lineWidth: 0.5))
    }
}

/// Compact last-10 over/under bar strip — the prop analog of the game card's
/// Polymarket sparkline. Green = cleared the line, red = missed. Bar height
/// scales with the actual value.
struct RecentFormStrip: View {
    let strip: [(cleared: Bool, value: Double)]
    let line: Double

    var body: some View {
        let maxVal = max(line * 1.5, (strip.map(\.value).max() ?? 1), 1)
        GeometryReader { geo in
            let n = max(strip.count, 1)
            let gap: CGFloat = 2
            let barW = max(2, (geo.size.width - gap * CGFloat(n - 1)) / CGFloat(n))
            HStack(alignment: .bottom, spacing: gap) {
                ForEach(Array(strip.enumerated()), id: \.offset) { _, bar in
                    let h = max(3, CGFloat(bar.value / maxVal) * geo.size.height)
                    RoundedRectangle(cornerRadius: 1.5)
                        .fill(bar.cleared ? Color.appPrimary : Color.appLoss.opacity(0.7))
                        .frame(width: barW, height: h)
                }
            }
            .frame(width: geo.size.width, height: geo.size.height, alignment: .bottom)
        }
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/props/components/PropPlayerCard.kt
package com.wagerproof.app.features.props.components

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.CornerRadius
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.hapticfeedback.HapticFeedbackType
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalHapticFeedback
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.app.features.props.PlayerPropSelection
import com.wagerproof.app.features.props.PlayerPropFeedItem
import com.wagerproof.app.features.props.PropsFormatting
import com.wagerproof.app.features.props.teamGlassDisc
import com.wagerproof.app.features.gamecards.teamVisible
import com.wagerproof.app.features.shared.RemoteImage
import com.wagerproof.app.features.shared.hexColor
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.models.MLBPlayerProps

/** A last-10 bar: cleared the line (green) or missed (red @0.7), height ∝ value. */
data class FormBar(val cleared: Boolean, val value: Double)

/**
 * MLB player-prop feed card — port of iOS `PropPlayerCard.swift`. Rounded lifted
 * surface, avatar-with-team-glow + team-logo chip, centered O/U pills, labeled
 * L10 trend strip, and a bottom BEST · L10 · HIT info row with a glass time pill.
 */
@Composable
fun PropPlayerCard(
    item: PlayerPropFeedItem,
    onSelect: (PlayerPropSelection) -> Unit,
    modifier: Modifier = Modifier,
) {
    val haptics = LocalHapticFeedback.current
    val sel = item.selection
    val computed = item.headline.computed
    val row = item.headline.row
    val primary = hexColor(item.teamPrimaryHex).teamVisible()
    val secondary = hexColor(item.teamSecondaryHex).teamVisible()
    val shape = RoundedCornerShape(26.dp)

    Column(
        modifier
            .fillMaxWidth()
            .clip(shape)
            .background(AppColors.appSurfaceElevated.copy(alpha = 0.55f))
            .border(0.5.dp, AppColors.appBorder.copy(alpha = 0.5f), shape)
            .clickable {
                haptics.performHapticFeedback(HapticFeedbackType.TextHandleMove)
                onSelect(sel)
            }
            .padding(start = 12.dp, end = 14.dp, top = 9.dp, bottom = 9.dp),
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        // Main row
        Row(verticalAlignment = Alignment.CenterVertically) {
            PropTeamAvatar(playerId = sel.playerId, teamLogoUrl = sel.teamLogoUrl, primary = primary, secondary = secondary)
            Spacer(Modifier.width(10.dp))
            Column {
                Text(
                    sel.playerName,
                    color = AppColors.appTextPrimary,
                    fontSize = 15.sp,
                    fontWeight = FontWeight.Bold,
                    maxLines = 1,
                )
                if (sel.opponentAbbr.isNotEmpty()) {
                    Text(
                        "vs ${sel.opponentAbbr}",
                        color = AppColors.appTextSecondary,
                        fontSize = 11.sp,
                        fontWeight = FontWeight.Medium,
                        maxLines = 1,
                    )
                }
            }
            Spacer(Modifier.weight(1f))
            Column(horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.spacedBy(4.dp)) {
                OuPill("O", MLBPlayerProps.formatLine(computed.line), MLBPlayerProps.formatOdds(computed.overOdds), AppColors.appPrimary)
                OuPill("U", MLBPlayerProps.formatLine(computed.line), MLBPlayerProps.formatOdds(computed.underOdds), AppColors.appTextSecondary)
            }
            Spacer(Modifier.width(10.dp))
            Column(horizontalAlignment = Alignment.End) {
                Text("L10 TREND", color = AppColors.appTextMuted, fontSize = 8.sp, fontWeight = FontWeight.Bold)
                Spacer(Modifier.height(3.dp))
                RecentFormStrip(
                    strip = computed.miniStrip.map { FormBar(it.cleared, it.value) },
                    line = computed.line,
                    modifier = Modifier.size(74.dp, 46.dp),
                )
            }
        }

        Box(Modifier.fillMaxWidth().height(0.5.dp).background(AppColors.appBorder.copy(alpha = 0.5f)))

        // Bottom info row
        Row(verticalAlignment = Alignment.CenterVertically) {
            InfoItem(item.metricLabel, MLBPlayerProps.marketLabel(row.market), AppColors.appPrimary)
            Spacer(Modifier.width(16.dp))
            InfoItem("L10", "${computed.l10.fractionLabel} Over", AppColors.appTextPrimary)
            Spacer(Modifier.width(16.dp))
            InfoItem("HIT", computed.l10.pctLabel, hitColor(computed.l10.pct))
            Spacer(Modifier.weight(1f))
            TimePill(PropsFormatting.gameTime(sel.gameTimeEt))
        }
    }
}

private fun hitColor(pct: Int?): Color {
    if (pct == null) return AppColors.appTextMuted
    return when {
        pct >= 70 -> AppColors.appPrimary
        pct >= 55 -> Color(0xFFEAB308)
        else -> AppColors.appTextSecondary
    }
}

/** Player headshot disc with a team-color glass ring + a small team-logo chip. */
@Composable
internal fun PropTeamAvatar(
    playerId: Int,
    teamLogoUrl: String?,
    primary: Color,
    secondary: Color,
) {
    Box(Modifier.size(44.dp), contentAlignment = Alignment.Center) {
        Box(Modifier.size(44.dp).teamGlassDisc(primary, secondary), contentAlignment = Alignment.Center) {
            PlayerHeadshot(playerId = playerId, size = 40.dp)
        }
        teamLogoUrl?.let { logo ->
            Box(
                Modifier
                    .align(Alignment.BottomEnd)
                    .size(20.dp)
                    .clip(CircleShape)
                    .background(AppColors.appSurfaceElevated)
                    .padding(2.dp),
            ) {
                RemoteImage(logo, null, Modifier.size(16.dp), contentScale = ContentScale.Fit)
            }
        }
    }
}

@Composable
internal fun PropTeamAvatarNFL(
    content: @Composable () -> Unit,
    teamLogoUrl: String?,
    primary: Color,
    secondary: Color,
) {
    Box(Modifier.size(44.dp), contentAlignment = Alignment.Center) {
        Box(Modifier.size(44.dp).teamGlassDisc(primary, secondary), contentAlignment = Alignment.Center) {
            content()
        }
        teamLogoUrl?.let { logo ->
            Box(
                Modifier
                    .align(Alignment.BottomEnd)
                    .size(20.dp)
                    .clip(CircleShape)
                    .background(AppColors.appSurfaceElevated)
                    .padding(2.dp),
            ) {
                RemoteImage(logo, null, Modifier.size(16.dp), contentScale = ContentScale.Fit)
            }
        }
    }
}

@Composable
internal fun OuPill(prefix: String, line: String, odds: String, tint: Color) {
    Row(
        Modifier
            .clip(CircleShape)
            .background(AppColors.appSurfaceMuted.copy(alpha = 0.55f))
            .border(0.5.dp, AppColors.appBorder.copy(alpha = 0.6f), CircleShape)
            .padding(horizontal = 8.dp, vertical = 4.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        Text("$prefix $line", color = AppColors.appTextMuted, fontSize = 9.sp, fontWeight = FontWeight.Bold)
        Text(odds, color = tint, fontSize = 11.sp, fontWeight = FontWeight.Bold, fontFamily = FontFamily.Monospace)
    }
}

@Composable
internal fun InfoItem(label: String, value: String, valueColor: Color) {
    Column {
        Text(label, color = AppColors.appTextMuted, fontSize = 8.sp, fontWeight = FontWeight.Bold)
        Text(value, color = valueColor, fontSize = 12.sp, fontWeight = FontWeight.SemiBold, maxLines = 1)
    }
}

@Composable
internal fun TimePill(text: String) {
    Box(
        Modifier
            .clip(CircleShape)
            .background(AppColors.appSurfaceElevated.copy(alpha = 0.92f))
            .border(0.5.dp, AppColors.appBorder.copy(alpha = 0.6f), CircleShape)
            .padding(horizontal = 7.dp, vertical = 3.dp),
    ) {
        Text(text, color = AppColors.appTextSecondary, fontSize = 9.sp, fontWeight = FontWeight.Bold, fontFamily = FontFamily.Monospace)
    }
}

/**
 * Compact last-10 over/under bar strip — the prop analog of the game card's
 * Polymarket sparkline. Green = cleared, red = missed, bar height ∝ actual
 * value. Hand-drawn with Compose Canvas (no external chart lib).
 */
@Composable
fun RecentFormStrip(
    strip: List<FormBar>,
    line: Double,
    modifier: Modifier = Modifier,
) {
    val maxVal = maxOf(line * 1.5, strip.maxOfOrNull { it.value } ?: 1.0, 1.0)
    Canvas(modifier) {
        val n = maxOf(strip.size, 1)
        val gap = 2.dp.toPx()
        val barW = maxOf(2f, (size.width - gap * (n - 1)) / n)
        strip.forEachIndexed { i, bar ->
            val h = maxOf(3f, (bar.value / maxVal).toFloat() * size.height)
            val x = i * (barW + gap)
            drawRoundRect(
                color = if (bar.cleared) AppColors.appPrimary else AppColors.appLoss.copy(alpha = 0.7f),
                topLeft = Offset(x, size.height - h),
                size = Size(barW, h),
                cornerRadius = CornerRadius(1.5.dp.toPx()),
            )
        }
    }
}
